draft-bbs-signatures.md

```diff
@@ -529,57 +532,59 @@ Outputs:
 
 Procedure:
 
-1. (A, e, s) = (octets_to_point(signature.A), OS2IP(signature.e), OS2IP(signature.s))
+1. signature_result = octets_to_signature(signature)
 
 2. (i1, i2,..., iR) = RevealedIndexes
 
 3. (j1, j2,..., jU) = [L] \ RevealedIndexes
 
-4. if subgroup_check(A) is INVALID, return INVALID
+4. if signature_result is INVALID, return INVALID
 
-5. if KeyValidate(PK) is INVALID, return INVALID
+5. (A, e, s) = signature_result
 
-6. generators =  (H_s || H_d || H_1 || ... || H_L)
+6. if KeyValidate(PK) is INVALID, return INVALID
 
-7. domain = OS2IP(hash(PK || L || generators || Ciphersuite_ID || header)) mod q
+7. generators =  (H_s || H_d || H_1 || ... || H_L)
 
-8. for element in (r1, r2, e~, r2~, r3~, s~, m~_j1, ..., m~_jU):
+8. domain = OS2IP(hash(PK || L || generators || Ciphersuite_ID || header)) mod q
 
-9.      element = hash(PRF(8*ceil(log2(q)))) mod q
+9. for element in (r1, r2, e~, r2~, r3~, s~, m~_j1, ..., m~_jU):
 
-10.      if element = 0, go back to step 7
+10.      element = hash(PRF(8*ceil(log2(q)))) mod q
 
-11. B = P1 + H_s * s + H_d * domain + H_1 * msg_1 + ... + H_L * msg_L
+11.      if element = 0, go back to step 7
 
-12. r3 = r1 ^ -1 mod q
+12. B = P1 + H_s * s + H_d * domain + H_1 * msg_1 + ... + H_L * msg_L
 
-13. A' = A * r1
+13. r3 = r1 ^ -1 mod q
 
-14. Abar = A' * (-e) + B * r1
+14. A' = A * r1
 
-15. D = B * r1 + H_s * r2
+15. Abar = A' * (-e) + B * r1
 
-16. s' = s + r2 * r3
+16. D = B * r1 + H_s * r2
 
-17. C1 = A' * e~ + H_s * r2~
+17. s' = s + r2 * r3
 
-18. C2 = D * (-r3~) + H_s * s~ + H_j1 * m~_j1 + ... + H_jU * m~_jU
+18. C1 = A' * e~ + H_s * r2~
 
-19. c = hash(PK || Abar || A' || D || C1 || C2 || ph)
+19. C2 = D * (-r3~) + H_s * s~ + H_j1 * m~_j1 + ... + H_jU * m~_jU
 
-20. e^ = e~ + c * e
+20. c = hash(PK || Abar || A' || D || C1 || C2 || ph)
```

Comment:
I think we’re fixing what goes into the hash in another PR right? If not I’ll create one that does

Reply:
Yeap we are